### DISCUSSION TITLE
cy.scope('#identifier')

### DISCUSSION THREAD

**Discussion by dillingham** [cypress-io/cypress#16572] [Component Testing] [answered]
would be handy if a before() hook could scope all upcoming `get()` by a selector

```js
before(() => {
    cy.visit('/page')
    cy.scope('#notifications');
})

it('can see all notifications', () => {
    cy.get('[test="trigger"]').click()

})
```
this selector
```js
cy.get('[test="trigger"]').click()
```
would become
```js
cy.get('#notifications [test="trigger"]').click()
```
behind the scenes

**Comment by jennifer-shehane** [comment, marked as answer]
This is similar to [`.within()`](https://on.cypress.io/within), but you can likely overwrite existing commands or write your own to get similar functionality: https://on.cypress.io/custom-commands

**Reply by dillingham** [reply]
@jennifer-shehane ah. didn't see `within`, thanks for the reply.
